pkgs/misc/scrcpy/default.nix

```diff
@@ -31,24 +31,15 @@ stdenv.mkDerivation rec {
     owner = "Genymobile";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-PWH+XLKraFfjXovnZpREXBaQVyOyP8yIMYDMiF6ddXg=";
+    sha256 = "sha256-M5zOKwqQ0y70gsI+c0Or7hUzz4fH/8DqcOeKq4Vryc4=";
   };
 
-  # Remove in the next patch release
-  patches = [
-    (fetchpatch {
-      name = "fix-macos-build-error.patch";
-      url = "https://github.com/Genymobile/scrcpy/commit/6b769675fa68e60c9765022e43c4d7b1e329353a.patch";
-      hash = "sha256-lQx01HI0nTWdZFusLIswZT2iOgkP84btqF6F58tGNko=";
-    })
-  ];
-
   # postPatch:
-  #   screen.c: When run without a hardware accelerator, this allows the command to continue working rather than failing unexpectedly.
+  #   display.c: When run without a hardware accelerator, this allows the command to continue working rather than failing unexpectedly.
   #   This can happen when running on non-NixOS because then scrcpy seems to have a hard time using the host OpenGL-supporting hardware.
   #   It would be better to fix the OpenGL problem, but that seems much more intrusive.
   postPatch = ''
-    substituteInPlace app/src/screen.c \
+    substituteInPlace app/src/display.c \
```

Comment:
Can we try to upstream this?

Reply:
I did end up talking about it to rom1v while doing this update and their opinion is that the software renderer should basically never be necessary.

> From https://github.com/Genymobile/scrcpy/issues/4105#issuecomment-1602765420
> > In scrcpy, it is not enabled, because if you get software rendering something is very wrong, and you should fix your configuration. Moreover, it seems it causes issues: [reddit.com/r/scrcpy/comments/14ezvex/my_screen_is_cut_when_running_scrcpy](https://www.reddit.com/r/scrcpy/comments/14ezvex/my_screen_is_cut_when_running_scrcpy/)

IME the patch does cause bugs, as can be seen [here](https://github.com/NixOS/nixpkgs/issues/230378)